hazelcast-code-generator/src/main/resources/codec-template-cpp.ftl

```diff
@@ -186,7 +186,7 @@ namespace hazelcast {
                     dataSize += ClientMessage::calculateDataSize<${util.getCppType(itemVariableType)} >(${varName});
                 <#break >
             <#case "ARRAY">
-                <#local itemVariableType= util.getGenericType(type)>
+                <#local itemVariableType= util.getArrayType(type)>
```

Comment:
Yes bug, build was failing because of that.
Nobody had used an array as parameter type, neither a List of Longs, so when I tried list, it failed too. Corrected both and checked the generated code, which compiles and looks ok.